internal/util/merr/utils.go

```diff
@@ -73,11 +73,17 @@ func Status(err error) *commonpb.Status {
 // Error returns a error according to the given status,
 // returns nil if the status is a success status
 func Error(status *commonpb.Status) error {
-	if status.Code == 0 {
+	if status.GetCode() == 0 && status.GetErrorCode() == commonpb.ErrorCode_Success {
 		return nil
 	}
 
-	return newMilvusError(status.GetReason(), status.Code, status.Code&retriableFlag != 0)
+	// use code first
+	code := status.GetCode()
+	if code == 0 {
+		code = int32(status.GetErrorCode())
```

Comment:
Can't use the old code as new code, use errUnexpect here

Reply:
fixed